### ISSUE TITLE
Connectivity broken in AWS CNI chaining mode

### ISSUE THREAD

**Issue by bmcustodio** [cilium/cilium#16007]
## Bug report

https://github.com/cilium/cilium/commit/f4431ca7c942fe67757cf013102da59ad1d36055 breaks connectivity when Cilium is running in AWS CNI chaining mode. As @pchaigno pointed out over Slack, using `--set endpointRoutes.enabled=true` seems to fix the issue, but as they also pointed out it might be better to not touch those routes if we're running in chaining mode to start with.

For reference, these are some example routes on a freshly created node without Cilium installed:

```
[root@ip-10-1-49-125 ~]# ip route
default via 10.1.32.1 dev eth0
10.1.32.0/19 dev eth0 proto kernel scope link src 10.1.49.125
10.1.51.245 dev eni6eb9ec1eefb scope link
10.1.57.21 dev enif1f8b0fd046 scope link
169.254.169.254 dev eth0
```

Also for reference, `10.1.51.245` and `10.1.57.21` were two CoreDNS pods. After Cilium was installed _without_ `--set endpointRoutes.enabled=true` those two routes were deleted.

**General Information**

- Cilium version (run `cilium version`)

  * `1.9.6`
  * `1.10.0-rc1`

- Kernel version (run `uname -a`)

    ```
    Linux ip-10-100-0-61.eu-west-1.compute.internal 5.4.110-54.182.amzn2.x86_64 #1 SMP Fri Apr 9 17:56:33 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
    ```

- Orchestration system version in use (e.g. `kubectl version`, ...)

    ```
    Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.8-eks-96780e", GitCommit:"96780e1b30acbf0a52c38b6030d7853e575bcdf3", GitTreeState:"clean", BuildDate:"2021-03-10T21:32:29Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
    ```

- Link to relevant artifacts (policies, deployments scripts, ...)

    N/A

- Generate and upload a system zip:

    N/A

**How to reproduce the issue**

    1. Follow the installation instructions for Cilium in AWS CNI chaining mode on a freshly created EKS cluster.

**Comment by pchaigno** [comment]
As said at the top, to fix it we should avoid [installing or removing endpoint routes](https://github.com/cilium/cilium/blob/3b64f2372319ec25c112c7c2d008d76b05ea8968/pkg/datapath/loader/loader.go#L355-L383) when we are in chaining mode. André suggested we could use the environment variable `CILIUM_CNI_CHAINING_MODE` to determine, in the agent, if we're running in chaining mode.

**Comment by joestringer** [comment]
There is a workaround for this issue and it is not a regression directly from v1.9.6, so I think we can defer this and fix it when it's ready. Unblocking so we can get the security release out.

**Comment by rewiko** [comment]
Same issue on our side.

**Comment by pchaigno** [comment]
@bmcustodio Could you try the fix at https://github.com/cilium/cilium/pull/16227? :pray:

**Comment by pchaigno** [comment]
The fix for this regression was released in v1.9.7 and will also be part of the upcoming v1.10.1 release.
